command/agent/auth/cert/cert.go

```diff
@@ -0,0 +1,65 @@
+package cert
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/command/agent/auth"
+)
+
+type certMethod struct {
+	logger    hclog.Logger
+	mountPath string
+	name      string
+}
+
+func NewCertAuthMethod(conf *auth.AuthConfig) (auth.AuthMethod, error) {
+	if conf == nil {
+		return nil, errors.New("empty config")
+	}
+
+	// Not concerned if the conf.Config is empty as the 'name'
+	// parameter is optional when using TLS Auth
+
+	c := &certMethod{
+		logger:    conf.Logger,
+		mountPath: conf.MountPath,
+		name:      "",
+	}
+
+	if conf.Config != nil {
+		nameRaw, ok := conf.Config["name"]
```

Comment:
Can we change the configuration parameter, and any matching variables, from `name` to `role` for clarity? I know the [parameter is "name"](https://www.vaultproject.io/api/auth/cert/index.html#name-6) but the documentation clarifies that it is the name of the role.

Reply:
I understand that using `role` over `name` in the agent configuration would make all of the methods look the same. Unfortunately, the TLS Authentication backend is different when it comes to interacting with them via the api or the cli. Also, I couldn't find the use of `role` as a variable name in the certificate authentication code either. The only references are in comments and the documentations about the concept of the _Certificate Role_. It feels most transparent to use the same name in both places as that would allow a curl test by hand and the agent configuration to look almost identical.